DEPRECATED.md

```diff
@@ -21,6 +21,9 @@ A logged warning is expected for each deprecated item that is in deprecation win
 * `SAN` is replaced by `URI` in the `x-forwarded-client-cert` header.
 * The `endpoint` field in the http health check filter is deprecated in favor of the `headers`
   field where one can specify HeaderMatch objects to match on.
+* The `hosts` field in the `Cluster` is deprecated in favor of the `load_assigment` field to accomodate
+  specifying `HealthCheckConfig` for each cluster member.
```

Comment:
It's quite a bit more than just the health check config, even if this was the original rationale :) Might be worth pointing out that CDS assignments can now contained embedded EDS equivalent endpoint assignments.

Reply:
@htuch yes, In general, I haven't touched the locality yet. Still trying to group some functionalities.